activerecord/lib/active_record/railtie.rb

```diff
@@ -178,6 +182,11 @@ class Railtie < Rails::Railtie # :nodoc:
     initializer "active_record.check_represent_sqlite3_boolean_as_integer" do
       config.after_initialize do
         ActiveSupport.on_load(:active_record_sqlite3adapter) do
+          sqlite3_represent_boolean_as_integer = Rails.application.config.active_record.delete(:sqlite3_represent_boolean_as_integer)
+          unless sqlite3_represent_boolean_as_integer.nil?
```

Comment:
Agree. 
In my feeling I think that it can be abstracted by `Configuration` class. But it seems that more consideration is needed. 

By the way, should I include it in this PR?